LibreNMS/Device/YamlDiscovery.php

```diff
@@ -338,13 +358,21 @@ public static function oidToNumeric($oid, $device = null, $mib = 'ALL', $mibdir
         if (self::oidIsNumeric($oid)) {
             return $oid;
         }
-
-        foreach (explode(':', $mib) as $mib_name) {
-            if ($numeric_oid = snmp_translate($oid, $mib_name, $mibdir, null, $device)) {
-                break;
+        $key = 'YamlDiscovery:oidToNumeric:' . $mibdir . '/' . $mib . '/' . $oid;
+        if (Cache::has($key)) {
+            $numeric_oid = Cache::get($key);
+        } else {
+            foreach (explode(':', $mib) as $mib_name) {
+                $numeric_oid = snmp_translate($oid, $mib_name, $mibdir, null, $device);
+                if ($numeric_oid) {
+                    break;
+                }
             }
         }
 
+        //Store the value
+        Cache::put($key, $numeric_oid, $cache_time);
```

Comment:
$this->cache_time ?

Reply:
Nope : 
`Using $this when not in object context`